RELEASE/scripts/autoscend/quests/level_07.ash

```diff
@@ -32,7 +32,10 @@ boolean L7_crypt()
 
 	void useNightmareFuelIfPossible()
 	{
-		if((spleen_left() > 0) && (item_amount($item[Nightmare Fuel]) > 0) && !is_unrestricted($item[Powdered Gold]))
+		//chews this when there are no guaranteed uses for spleen 
```

Comment:
This should check to see if we are out of spleen familiar _drops_
we might have for example the monkey that gives 1 drop per day worth 4 spleen size.
our spleen could be sitting at 4/15 with the 1 spleen drop already chewed up. and we will refuse to chew up the remaining nightmare fuels.

So I think instead of `haveSpleenFamiliar()` it should be a check for quantity of available and valid adv giving spleen drops.
something like
```
int spleen_food_count;
foreach it in $items[list of items goes here]
{
spleen_food_count++;
}
```
and then replace `haveSpleenFamiliar()` with `spleen_food_count > 0`

as for the possibility it did not drop yet... we actually give pretty good priority to such familiars so if it was going to drop it dropped already most likely.

Reply:
monkey gives more than 1 drop in QT. I think all drop as much as you can put in a spleen

but this will still be useful when QT is out of standard and you could have nightmare fuel and spleen items without familiar

Reply:
my point stands though, it should be checking for `spleen_food_count > 0` rather than the existence of a pet.
... actually in QT the check is even more meaningless since you have all the pets but also have no access to them so haveSpleenFamiliar() is meaningless question to ask. It either always returns true in QT, or it always returns "is current familiar a spleen familiar"

Reply:
yes I meant what you said will be useful out of standard for that reason, also there is some spleen left after 3x4